driver/network_driver.go

```diff
@@ -252,8 +290,12 @@ func (d NetworkDriver) Join(request *network.JoinRequest) (*network.JoinResponse
 		return nil, err
 	}
 
+	mac, err := d.GetEndpointMAC(request.EndpointID)
+	if err != nil {
+		return nil, err
+	}
 	// libnetwork doesn't set the MAC address properly, so set it here.
```

Comment:
I'm not happy merging a change that introduces an extra round trip to the datastore. So I think it would be better to remove the netns.SetVethMac as part of this PR. It would also make this PR much smaller, which is a good thing!